### PR TITLE
[mdatagen] Generate tests for the code generated by metrics builder

### LINKED ISSUE #16375: [Metrics builder] Generate tests to cover the generated code
Generate `internal/metadata/generated_metrics_test.go`

### PR DESCRIPTION
**Pull request by dmitryax**
There is some more functionality planned to be added to the metrics builder. Before proceeding with that, we need to cover the generated code with unit tests.

Changes are made to `cmd/mdatagen` only. Everything else is auto-generated

Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16375

### CODE REVIEW

File: cmd/mdatagen/loader.go [hunk 1]
```diff
@@ -79,6 +79,11 @@ func (mvt *ValueType) UnmarshalText(text []byte) error {
 
 // String returns capitalized name of the ValueType.
 func (mvt ValueType) String() string {
+	// Temporary assume that empty value type is string because "type" is an optional field for data point attributes.
+	// TODO: Make the "type" attribute a required field for data point attributes as it's done for resource attributes.
```
Comment (djaglowski) [inline cmd/mdatagen/loader.go:83]: Noting that this is solved when https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16420 is merged.

**Review by djaglowski — COMMENTED** [review]
LGTM, assuming we wait for the noted todo to be resolved by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16420

File: cmd/mdatagen/loader.go (hunk 1 above)
Comment (dmitryax) [reply cmd/mdatagen/loader.go]: Resolved

**Comment by dmitryax** [comment]
@djaglowski, rebased. PTAL

**Review by djaglowski — APPROVED** [review]

**Pull request merged**